docker/supabase/volumes/db/init/04-telescope-schema.sql

```diff
@@ -39,3 +39,36 @@ CREATE TRIGGER set_timestamp
 BEFORE UPDATE ON telescope_profiles
 FOR EACH ROW
 EXECUTE PROCEDURE trigger_set_timestamp();
+
+-- Tables policies that restrict, on request basic, which rows can be returned from a query
+-- Note: servers using SERVICE_ROLE_KEY, are not restricted by RLS
+
+ALTER TABLE telescope_profiles ENABLE ROW LEVEL SECURITY;
+
+-- Disable all operations, unless being overwritten by other policies
+CREATE POLICY profiles_policy ON telescope_profiles
+    FOR ALL
+    USING (false);
+
+-- Enable read access for Seneca users
+CREATE POLICY profiles_read_policy ON telescope_profiles
+    FOR SELECT
+    USING (((current_setting('request.jwt.claims'::text, true))::json ->> 'email'::text) IS NOT NULL);
```

Comment:
A better claim here would be to use the `role`.  We (currently) set the `role` to an `Array`, and it will be one of:

- `["seneca"]`
- `["seneca", "telescope"]`
- `["seneca", "telescope", "admin"]`

Either you should check to see if `role` includes `"seneca"` (not sure how you do that), or we can also play with how we define the role to be a string (e.g., `role="telescope"` or `role="seneca"`).  

I don't think there's any value in the `"admin"` role, since we're going to use GitHub team membership instead.

Reply:
I might need your help on this if you can, I didn't manage to write a valid SQL expression to check if a role is present in the role array.

If you go to http://localhost:8910/project/default/auth/policies, to add a new policy, the SQL expression for that would look something like this
```
('seneca' in ((current_setting('request.jwt.claims'::text, true))::json #> '{role}'::text[]))
```
But that's an invalid expression.
in this case though, I think having an email is already sufficient

Reply:
I think we should try to solve this, and I agree it's not obvious (to me) how to do it.  I can't believe how little there is in the way of code examples online for this problem.

Based on what Supabase does, I suggest we write our own function(s) to help us accomplish this.  I want to write something like this (but it's basically pseudo-code):

```sql
create or replace function user_has_role(role text)
returns boolean
language sql stable
as $$
  IF (role IN current_setting('request.jwt.claim.roles', true))
    then return true;
  else
    return false;
  END IF;
$$;
```

Then we can use `user_has_role("seneca")` or `user_has_role("telescope")`

@JerryHue would you be willing to help out here?  We need a way to be able to extract the `Array` of `roles` (`string[]`) from the JWT, and check if a particular value is included.  I *think* using `IN` is the right way, but again, I've spent my career avoiding databases as much as possible!

Reply:
For reference, here are examples of the kind of functions that Supabase is using that I want to copy:

- https://github.com/Seneca-CDOT/telescope/blob/master/docker/supabase/volumes/db/init/01-auth-schema.sql#L92-L123
- https://github.com/Seneca-CDOT/telescope/blob/master/docker/supabase/volumes/db/init/02-storage-schema.sql#L37-L93

Reply:
If this is going to block us, I'm also willing to change our `roles` from being `string[]` to just `string`:

- `"roles": ["seneca"]` would become `role: "seneca"`
- `"roles": ["seneca", "telescope"]` would become `role: "telescope"`

If we do that, the existing `auth.role()` function would work right now.

Reply:
Aright, I figured it out by luck. The key is to use [jsonb](https://www.postgresql.org/docs/current/datatype-json.html#JSON-CONTAINMENT) and its containment check operator. 
```sql
(((current_setting('request.jwt.claims'::text, true))::jsonb #> '{role}'::text[]) ? 'seneca'::text)
```

Reply:
That converts a jwt into a `jsonb`, gets the value of the `roles`, which is a JSON array, and checks if "seneca" is in that array.